src/docfx/lib/markdown/IncludeExtension.cs

```diff
@@ -65,6 +65,12 @@ private static void RemoveIncludeRenderer(IMarkdownRenderer renderer)
                 return;
             }
 
+            if (!inclusionBlock.IncludedFilePath.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
```

Comment:
Move this check before calling `_context.ReadFile`

Reply:
Moved.